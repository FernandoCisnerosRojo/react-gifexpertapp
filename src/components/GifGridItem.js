import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({ url, title }) => {

    return (
        <div className="card animate__animated animate__flipInX">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default GifGridItem
