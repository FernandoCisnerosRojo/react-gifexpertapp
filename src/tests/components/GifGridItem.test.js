import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import GifGridItem from '../../components/GifGridItem';

const title = 'titleTest'
const url = 'urlTest'
const wrapper = shallow(<GifGridItem url={url} title={title} />);

describe('GifGridItem Tests', () => {
    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should render title in p tag', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title);
    })

    test('should render image from url with title as alt property', () => {
        const img = wrapper.find('img')

        // console.log(img.props());
        // console.log(img.prop('src'));
        // console.log(img.prop('alt'));

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should have animate__flipInX css class', () => {
        const div = wrapper.find('div')

        const className = div.prop('className');

        expect(className.includes('animate__flipInX')).toBe(true);
    })

})
