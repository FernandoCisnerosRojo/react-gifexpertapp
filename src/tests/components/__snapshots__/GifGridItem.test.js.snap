// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Tests should match snapshot 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="titleTest"
    src="urlTest"
  />
  <p>
    titleTest
  </p>
</div>
`;
